utility/src/main/java/org/oppia/android/util/networking/DebugNetworkConnectionUtil.kt

```diff
@@ -0,0 +1,45 @@
+package org.oppia.android.util.networking
+
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * [NetworkConnectionUtil] that gets and sets the current [ConnectionStatus] of the device in debug builds.
+ */
+@Singleton
+class DebugNetworkConnectionUtil @Inject constructor(
+  private val prodNetworkConnectionUtil: ProdNetworkConnectionUtil
+) : NetworkConnectionUtil {
+
+  private var forcedConnectionStatus: ConnectionStatus? = null
+
+  override fun getCurrentConnectionStatus(): ConnectionStatus {
+    val actualConnectionStatus = prodNetworkConnectionUtil.getCurrentConnectionStatus()
+    if (actualConnectionStatus == ConnectionStatus.NONE) {
+      forcedConnectionStatus = null
+    }
```

Comment:
This is a bit unexpected--usually getters don't have side-effects (except in rare cases when we want to use something like memoization which doesn't have _logical_ side effects). Why is this correct?

Reply:
I have added this check to prevent impossible scenarios. Consider an example where the user is connected to Cellular network and has forced the app to use Wifi network. Now, the user turns off the network in his device. This basically means that the device now has no connection but we are forcing a Wifi network -- this is an impossible scenario. So, this check will set the network type to default automatically in the case the previously mentioned scenario happens. Does this make sense?

Reply:
Yes, that makes sense, though I would actually expect the app to try to use wifi since the user instructed it to. It shouldn't actually work when they try to download something, but it's actually a useful scenario: it's testing when the app thinks it has wifi but connectivity doesn't work (which definitely happens to people). I suggest reverting this special case since it will simplify the implementation & tests, and allow us to test those sorts of scenarios.

Otherwise, if a user breaks the app because they combine these contexts then they can always change back the debug settings to follow the system default.

Reply:
Removed extra checks. PTAL.